idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/lineMarker/LineMarkerSyntax.kt

```diff
@@ -115,4 +199,38 @@ interface LineMarkerSyntax {
           override fun getClickAction(): AnAction? = clickAction
         }
     }
+
+  fun <R> LineMarkerSyntax.navigationGutter(
+    icon: Icon,
+    element: PsiElement,
+    config: NavigationGutterIconBuilder<PsiElement>.(element: PsiElement) -> R
+  ): R =
+    NavigationGutterIconBuilder.create(icon).config(element)
+
+  /**
+   * `MergeableLineMarkerInfo` can merge multiple LineMarkerInfo's into one, if [mergeWith] is true.
+   * @param commonIcon defines the common Icon after the merge
+   */
+  fun LineMarkerSyntax.mergeableLineMarkerInfo(
+    icon: Icon,
+    element: PsiElement,
+    message: (PsiElement) -> String,
+    commonIcon: MergeableLineMarkerInfo<PsiElement>.(others: List<MergeableLineMarkerInfo<PsiElement>>) -> Icon = { icon },
+    mergeWith: MergeableLineMarkerInfo<PsiElement>.(other: MergeableLineMarkerInfo<*>) -> Boolean = { this.icon == it.icon },
+    placed: GutterIconRenderer.Alignment = GutterIconRenderer.Alignment.LEFT,
+    navigate: (event: MouseEvent, element: PsiElement) -> Unit = Noop.effect2,
+    clickAction: AnAction? = null
+  ): MergeableLineMarkerInfo<PsiElement> =
```

Comment:
would it make sense to use a type parameter for the `PsiElement`? This could be used for parameters 'message`, `commonIcon`, `mergeWith`, and perhaps `navigate`. The return type would also take this type parameter.

Reply:
that is already provided in `addLineMarkerProviderM`